api/v1/checks.go

```diff
@@ -779,6 +779,23 @@ func (c GitHubCheck) GetEndpoint() string {
 	return strings.ReplaceAll(c.Query, " ", "-")
 }
 
+type GitProtocolCheck struct {
+	Description    `yaml:",inline" json:",inline"`
+	Templatable    `yaml:",inline" json:",inline"`
+	ConnectionName string       `yaml:"connection,omitempty" json:"connection,omitempty"`
+	Repository     string       `yaml:"repository" json:"repository"`
+	Username       string       `yaml:"username" json:"username"`
```

Comment:
The convention we use elsewhere is that both the username and password are of types.EnvVar, can you change this to be more consistent ?

Reply:
Done